### DISCUSSION TITLE
I strongly recommended adding upload functionality, and I was disappointed that it wasn't available

### DISCUSSION THREAD

**Discussion by ReanRelay** [navidrome/navidrome#3412] [Ideas]
For example, if you are listening to your favorite songs on other players and want to upload the songs to your own server, if you log in to the background, use or add other apps to upload, it is a very stupid and troublesome behavior

**Comment by kgarner7** [comment]
Navidrome is designed to be read-only for your music. It is very unlikely for this to change.

**Reply by ReanRelay** [reply]
You know what? You might as well write your own music player

**Comment by Fabian-G** [comment]
I actually prefer it this way, because there are other tools, which are way better at managing files. Since you can mount your Music directory read-only into navidrome you can use whatever to upload files. For example just upload using nextcloud and mount that directory ro into navidrome. Or personally, if I want to add music, I mount my music directory locally using sshfs and then import with beets.

**Comment by gadgetvala** [comment]
I agree with @ReanRelay  — it should have an option to upload and delete music directly from the player. Since I use it on my home network and mobile phone, it would be more convenient to upload songs directly from the browser instead of manually adding them to the folder by logging into the network.

I've shared access with my friends to build a larger music library. A general music upload feature would be really helpful, where any user can upload songs, but only the song owner or admin can manage them. Other users should still be able to search for those songs and add them to their playlists.

I believe this would be a valuable and helpful feature to add.

**Comment by bh90210** [comment]
![image](https://github.com/user-attachments/assets/f18479de-5f91-4814-8b8f-f8e95937c764)
I actually needed this feature so locally I have it, along with download. So there is demand and implementations in the wild

**Comment by MRodneyMcKay** [comment]
By design, navidrome do not modify files, so it would be pointless to make it so. But honestly, I do not see the usecase for it. Instead of logging into webplayer, you could just copy to a samba share or just straight download to the dataset
